e3/os/process.py

```diff
@@ -302,12 +302,21 @@ def add_interpreter_command(cmd_line):
         self.err = ''
         self.cmds = []
 
-        if env is not None and not ignore_environ:
-            # ignore_environ is False, so get a copy of the current
-            # environment and update it with the env dictionnary.
-            tmp = os.environ.copy()
-            tmp.update(env)
-            env = tmp
+        if env is not None:
+            if ignore_environ:
+                if sys.platform == 'win32':
+                    tmp = {}
```

Comment:
I would move the (excellent!) comment you provided in the revision directly in the code, somewhere here.